src/BuiltInTools/dotnet-format/Logging/SimpleConsoleLogger.cs

```diff
@@ -1,18 +1,18 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
 
 using System.Collections.Immutable;
-using System.CommandLine;
-using System.CommandLine.Rendering;
 using Microsoft.Extensions.Logging;
 
 namespace Microsoft.CodeAnalysis.Tools.Logging
 {
     internal class SimpleConsoleLogger : ILogger
     {
-        private readonly object _gate = new object();
+        private static readonly bool ColorsAreSupported =
+            !(OperatingSystem.IsBrowser() || OperatingSystem.IsAndroid() || OperatingSystem.IsIOS() || OperatingSystem.IsTvOS())
+                && !Console.IsOutputRedirected;
+
+        private readonly static object _gate = new object();
```

Comment:
I made the gate static, as `Console` is a static singleton and I expect that we want to prevent from multiple threads changing the color.